### ISSUE TITLE
The "Completion provider example" does not work

### ISSUE THREAD

**Issue by adudsic** [microsoft/monaco-editor#1294]
In the playground web site, the "Completion provider example" does not work.

**TL;DR:** The JSON language provider used on the web site is not the current version. The online version (v2.3.0) contains a bug (missing a check for null) that has already been fixed in the JSON language provider repo (v3.3.0).

**REPRO STEPS:**
1. Navigate to 
[https://microsoft.github.io/monaco-editor/playground.html#extending-language-services-completion-provider-example](https://microsoft.github.io/monaco-editor/playground.html#extending-language-services-completion-provider-example)
2. Click **Run** and then click line 3.
3. Type a double-quote and then the letter "L".
EXPECTED RESULT: Editor should display a hover box containing "lodash".
ACTUAL RESULT: No hover box displayed.

4. Reload the page.
5. In line 27, change the first argument ('json') to: 'xml'.
6. In line 41, change the value of language to: 'xml'.
7. Click **Run** and then click line 3.
8. Type a double-quote and then the letter "L".
RESULT: Editor displays a hover box containing "lodash".
IMPLICATION: Issue is not with the **CompletionItemProvider**, but instead has something to do with the language.

9. Reload the page (i.e., back to using JSON).
10. In line 40, replace the value of value with: "\\"\\""
11. Open the browser's Dev tools console (**F12** or **Ctrl+Shift+I**).
12. Click **Run**. 
13. In the editor, hover your mouse over the double-quotes.
RESULT: Dev Console error stack shows: 
```
   TypeError: Cannot read property 'type' of null  
        at e.doHover (jsonWorker.js:7)  
        at e.doHover (jsonWorker.js:7)
```

**DETAILS:** jsonWorker.js creates a variable and assigns the value of "c.parent", and in this scenario, "c.parent" is _null_. The fix would be to check for _null_.

I looked through the VS Code source in GitHub, and found that the code implementing **doHover()** is imported from the ‘vscode-json-langaugeservice’; specifically, from the **JSONHover** class defined here: 
[https://github.com/Microsoft/vscode-json-languageservice/blob/master/src/services/jsonHover.ts](https://github.com/Microsoft/vscode-json-languageservice/blob/master/src/services/jsonHover.ts)

Apparently, though, the null check has already been implemented on Aug 15, 2018. In fact, the repro steps listed in the issue [Hover fails on json file containing only string]https://github.com/Microsoft/vscode-json-languageservice/issues/25) describe how I reproduced the bug.

In the playground, if you reproduce the error (steps 9-13 above) and click through on jsonWorker.js, the head of the file indicates: ```monaco-json version: 2.3.0```

It looks like the Git repro for vscode-json-languageservice is on version 3.3.0.

**CONCLUSION:** The playground's "Completion provider example" does not work as-is using the provided JSON example because the code hits a null error. The source of the bug has been fixed in the underlying vscode-json-languageservice code, but this updated code is not currently being used by the playground web site.

**Comment by ProductiveRage** [comment]
@adudsic - no wonder I couldn't work out what behaviour that example was supposed to illustrate! I tried making the changes to the code that you described and now I can see it working. Thanks!

**Comment by aeschli** [comment]
I updated the monaco-json dependencies.
